modules/standard/IO.chpl

```diff
@@ -7067,6 +7070,9 @@ proc channel.skipField() throws {
   :throws SystemError: Thrown if the string could not be formatted.
  */
 proc string.format(args ...?k): string throws {
+  // TODO: Without the following line, I get a deprecation warning
+  // for SysBasic, yet nothing seems to be importing/using it...?!?
+  import OS.POSIX.{EILSEQ,EINVAL};
```

Comment:
@lydia-duncan : I wonder whether you would have a few spare minutes to help me out with this puzzle.  What I'm finding is that without this `import` statement, I get my deprecation warning from SysBasic.EILSEQ, SysBasic.EINVAL for a simple hello.chpl compile.  Yet I don't see any way in which this routine could even see those two symbols from SysBasic.  Specifically:
* the import statements from this file for SysBasic filter down to a very small set of symbols not including these two
* all other uses/imports of SysBasic in other modules in the module directory are private
* I'm not seeing any compiler code that would inject a use or import of SysBasic

I'd like to think that I'm just missing something simple or obvious, but am worried that it's indicative of a namespace-related bug.

Reply:
Hmm, that is interesting.  The function _is_ generic, so my best guess is that it could be getting instantiated someplace where SysBasic is used and then resolving the symbol references there?

Reply:
Oh, that's a great insight.  Let me look into that, thanks!

Reply:
Oh, but wait:  Don't the deprecation warnings get printed out in scopeResolve before any generic instantiation is done?

Reply:
Yeah, looks like it is getting resolved during scopeResolution.  Weird.  Lemme dig a little bit

Reply:
Got it - it's because we're in a string method.  We go to where the string type is defined to look for symbols, and the String module has a use of SysBasic

Reply:
Ah, so not a result of being generic so much as a tertiary method?  Thank you!

Reply:
I suspect it's more an oversight than intentional, tbh.  Function resolution doesn't follow private uses or imports when jumping to the type's definition, but it doesn't look like that part of scope resolution is set up to notice we got to that scope by jumping rather than traversing up scopes.

Reply:
(I was uncertain when I found the source of it whether it was okay, but thinking about it more, I agree with you)

Reply:
OK, I'll open up an issue for it.

Reply:
I've created #19883 for this.